localstack/services/awslambda/lambda_executors.py

```diff
@@ -783,14 +794,53 @@ def execute_in_container(
                 container_info.name, f"{lambda_cwd}/.", DOCKER_TASK_FOLDER
             )
 
-        return DOCKER_CLIENT.exec_in_container(
-            container_name_or_id=container_info.name,
-            command=inv_context.lambda_command,
-            interactive=True,
-            env_vars=env_vars,
-            stdin=stdin,
+        lambda_docker_ip = DOCKER_CLIENT.get_container_ip(container_info.name)
+
+        if not lambda_docker_ip or not in_docker():
```

Comment:
Note: Over time we could add an additional check here that verifies whether the target endpoint (`f"http://{lambda_docker_ip}:9001"`) is actually accessible in `invoke_lambda(..)` below, and otherwise also fall back to `exec_in_container(..)`. 

This could help deal with issues where potentially the Lambda container is not accessible (e.g., because it's running in a different Docker network). Should usually not be the case (because Lambdas typically also call back to the LocalStack APIs on the same network), but technically possible..